aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -745,7 +751,8 @@ SparseTensor& mul_out_sparse_cpu(const Tensor& t_, const Tensor& src_, SparseTen
   AT_ASSERT(!t_.is_cuda()); // dispatch argument
   TORCH_CHECK(!r.is_cuda(), "mul: expected 'out' to be CPU tensor, but got CUDA tensor");
   TORCH_CHECK(!src_.is_cuda(), "mul: expected 'other' to be a CPU tensor, but got a CUDA tensor");
-
+  TORCH_CHECK(src_.is_sparse(), "mul(sparse, dense) is not supported");
+  TORCH_CHECK(t_.is_sparse(), "mul(dense, sparse) is not supported");
```

Comment:
Probably the same checks are needed in `mul_out_sparse_cuda` from `aten/src/ATen/native/sparse/cuda/SparseCUDATensorMath.cu`.

Reply:
Even when I add these here I get a different error, must be from earlier in the call chain:
> NotImplementedError: Could not run 'aten::is_coalesced' with arguments from the 'CUDA' backend. This could be because the operator doesn't exist for this backend, or was omitted during the selective/custom build process (if using custom build)

I did add a test that checks `NotImplementedError`. `TORCH_CHECK` becomes a `RuntimeError`.

The original issue was around the `INTERNAL ASSERT FAILED at /pytorch/aten/src/ATen/native/sparse/SparseTensorMath.cpp:69, please report a bug to PyTorch.` message for the error, and is fixed by the PR in its current state.